framework/Models/EnsembleModel.py

```diff
@@ -510,37 +499,6 @@ def evaluateSample(self, myInput, samplerType, kwargs):
     return returnValue
 
   def submit(self,myInput,samplerType,jobHandler,**kwargs):
-    """
-      This will submit an individual sample to be evaluated by this model to a
-      specified jobHandler. Note, some parameters are needed by createNewInput
-      and thus descriptions are copied from there.
-      @ In, myInput, list, the inputs (list) to start from to generate the new one
-      @ In, samplerType, string, is the type of sampler that is calling to generate a new input
-      @ In,  jobHandler, JobHandler instance, the global job handler instance
-      @ In, **kwargs, dict,  is a dictionary that contains the information coming from the sampler,
-        a mandatory key is the sampledVars'that contains a dictionary {'name variable':value}
-      @ Out, None
-    """
-    for mm in utils.returnImportModuleString(jobHandler):
-      if mm not in self.mods:
-        self.mods.append(mm)
-
-    prefix = kwargs['prefix']
-    self.tempOutputs['uncollectedJobIds'].append(prefix)
-
-    ## Ensemble models need access to the job handler, so let's stuff it in our
-    ## catch all kwargs where evaluateSample can pick it up, not great, but
-    ## will suffice until we can better redesign this whole process.
-    kwargs['jobHandler'] = jobHandler
-
-    ## This may look a little weird, but due to how the parallel python library
-    ## works, we are unable to pass a member function as a job because the
-    ## pp library loses track of what self is, so instead we call it from the
-    ## class and pass self in as the first parameter
-    jobHandler.addJob((self, myInput, samplerType, kwargs), self.__class__.evaluateSample, prefix, kwargs)
-
-
-  def submitAsClient(self,myInput,samplerType,jobHandler,**kwargs):
```

Comment:
Why can these two methods be combined now?  Did something get changed at the job handling level in the rework?

Reply:
For my understanding, Ensemble model always use ```submitAsClient``` in the past, now the pr removed the ```if else``` conditions in the step, that's why we can combine this two method.

Reply:
Yep. @wangcj05 is right. This specialization in the Step was introduced but it is not needed at all since the EnsembleModel is always run in Client mode.